packages/jest-runtime/src/index.js

```diff
@@ -509,6 +509,10 @@ class Runtime {
       ({
         enumerable: true,
         get() {
+          if (localModule.filename === from) {
```

Comment:
I originally had `moduleRegistry[from] === localModule`, but I think it's the same thing in practice